packages/@aws-cdk/aws-codedeploy/lib/lambda/deployment-group.ts

```diff
@@ -0,0 +1,282 @@
+   */
+  deploymentGroupName?: string;
+
+  /**
+   * The Deployment Configuration this Deployment Group uses.
+   *
+   * @default LambdaDeploymentConfig#AllAtOnce
+   */
+  deploymentConfig?: ILambdaDeploymentConfig;
+
+  /**
+   * The CloudWatch alarms associated with this Deployment Group.
+   * CodeDeploy will stop (and optionally roll back)
+   * a deployment if during it any of the alarms trigger.
+   *
+   * Alarms can also be added after the Deployment Group is created using the {@link #addAlarm} method.
+   *
+   * @default []
+   * @see https://docs.aws.amazon.com/codedeploy/latest/userguide/monitoring-create-alarms.html
+   */
+  alarms?: cloudwatch.Alarm[];
+
+  /**
+   * The service Role of this Deployment Group.
+   *
+   * @default a new Role will be created.
+   */
+  role?: iam.Role;
+
+  /**
+   * Lambda Alias to shift traffic. Updating the version
+   * of the alias will trigger a CodeDeploy deployment.
+   */
+  alias: lambda.Alias;
+
+  /**
+   * The Lambda function to run before traffic routing starts.
+   */
+  preHook?: lambda.IFunction;
+
+  /**
+   * The Lambda function to run after traffic routing starts.
+   */
+  postHook?: lambda.IFunction;
+
+  /**
+   * Whether to continue a deployment even if fetching the alarm status from CloudWatch failed.
+   *
+   * @default false
+   */
+  ignorePollAlarmsFailure?: boolean;
+
+  /**
+   * The auto-rollback configuration for this Deployment Group.
+   */
+  autoRollback?: AutoRollbackConfig;
+}
+
+export class LambdaDeploymentGroup extends cdk.Construct implements ILambdaDeploymentGroup {
+  /**
+   * Import an Lambda Deployment Group defined either outside the CDK,
+   * or in a different CDK Stack and exported using the {@link #export} method.
+   *
+   * @param parent the parent Construct for this new Construct
+   * @param id the logical ID of this new Construct
+   * @param props the properties of the referenced Deployment Group
+   * @returns a Construct representing a reference to an existing Deployment Group
+   */
+  public static import(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupImportProps): ILambdaDeploymentGroup {
+    return new ImportedLambdaDeploymentGroup(scope, id, props);
+  }
+
+  public readonly application: ILambdaApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+
+  private readonly serviceRole: iam.Role;
+  private readonly alarms: cloudwatch.Alarm[];
+  private preHook?: lambda.IFunction;
+  private postHook?: lambda.IFunction;
+
+  constructor(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupProps) {
+    super(scope, id);
+    this.application = props.application || new LambdaApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    let serviceRole: iam.Role | undefined = props.role;
+    if (serviceRole) {
+      if (serviceRole.assumeRolePolicy) {
+        serviceRole.assumeRolePolicy.addStatement(new iam.PolicyStatement()
```

Comment:
The "right" thing to do here is to add support in `PolicyDocument` to conditionally add a statement if a statement with the same SID was not already added and then use some UUID as a SID to avoid duplicates.